contracts/modules/Wallet/VestingEscrowWallet.sol

```diff
@@ -0,0 +1,525 @@
+        templates.push(template);
+        /*solium-disable-next-line security/no-block-members*/
+        emit AddTemplate(_numberOfTokens, _duration, _frequency, now);
+    }
+
+    /**
+     * @notice Removes template
+     * @param _index index of the template
+     */
+    function removeTemplate(uint256 _index) external withPerm(ADMIN) {
+        require(_index < templates.length, "Template not found");
+        templates[_index] = templates[templates.length - 1];
+        templates.length--;
+        /*solium-disable-next-line security/no-block-members*/
+        emit RemoveTemplate(_index, now);
+    }
+
+    /**
+     * @notice Returns count of templates
+     * @return count of templates
+     */
+    function getTemplateCount() external view withPerm(ADMIN) returns(uint256) {
+        return templates.length;
+    }
+
+    /**
+     * @notice Adds vesting schedules for each of beneficiary
+     * @param _beneficiary beneficiary's addresses
+     * @param _numberOfTokens number of tokens
+     * @param _duration vesting duration
+     * @param _frequency vesting frequency
+     * @param _startTime vesting start time
+     */
+    function addSchedule(
+        address _beneficiary,
+        uint256 _numberOfTokens,
+        uint256 _duration,
+        uint256 _frequency,
+        uint256 _startTime
+    )
+        public
+        withPerm(ADMIN)
+    {
+        _validateSchedule(_beneficiary, _numberOfTokens, _duration, _frequency, _startTime);
+        require(_numberOfTokens <= unassignedTokens, "Not enough tokens");
+
+        Schedule memory schedule;
+        unassignedTokens = unassignedTokens.sub(_numberOfTokens);
+        schedule.numberOfTokens = _numberOfTokens;
+        schedule.lockedTokens = _numberOfTokens;
+        schedule.duration = _duration;
+        schedule.frequency = _frequency;
+        schedule.startTime = _startTime;
+        schedule.nextTime = _startTime.add(schedule.frequency);
+        schedule.state = State.CREATED;
+        //add beneficiary to the schedule list only if adding first schedule
+        if (dataMap[_beneficiary].schedules.length == 0) {
+            dataMap[_beneficiary].index = beneficiaries.length;
+            beneficiaries.push(_beneficiary);
+        }
+        dataMap[_beneficiary].schedules.push(schedule);
+        /*solium-disable-next-line security/no-block-members*/
+        emit AddSchedule(_beneficiary, _numberOfTokens, _duration, _frequency, _startTime, now);
+    }
+
+    /**
+     * @notice Adds vesting schedules from template for each of beneficiary
+     * @param _beneficiary beneficiary's addresses
+     * @param _index index of the template
+     * @param _startTime vesting start time
+     */
+    function addScheduleFromTemplate(address _beneficiary, uint256 _index, uint256 _startTime) public withPerm(ADMIN) {
+        require(_index < templates.length, "Template not found");
+        Template storage template = templates[_index];
```

Comment:
you can use memory here as you are not changing the value of the Template. and function could be external